### PR TITLE
Add documentation for dry-run setup from build-config.yaml

### PR DESCRIPTION
**Pull request by jomrazek**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/bacon/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: guide/build-config.md [hunk 1]
```diff
@@ -69,13 +70,16 @@ milestone: DR1
...
 
 The `releaseStorageUrl` is location of the release storage, typically on rcm-guest staging. This is the url to be used in the 'upload to candidates script' when running the `pig release` command. Using `*` in the milestone means that this property may also be used for automatic increment of milestone value. For example `ER*` as milestone, during `pig run`, the command will check the configured release storage and increment the version to next available milestone version. It is possible to configure it either within the `build-config.yaml` or via the CLI using the flag `--releaseStorageUrl=<...>` with the former taking precedence.
 
 The `brewTagPrefix` is an **optional** key that you can use to override the default generated brew tag prefix associated with a product version. The key is used to specify to which Brew tag all the builds should go when the milestone is closed.
 
+The `temporaryBuildAlignmentPreference` is option to specify alignment preference for temporary build to be run with: PERSISTENT or TEMPORARY. This property will override temporary or persistent flags passed in as command line options.
```
Comment (thescouser89) [inline guide/build-config.md:81]: "is an option"

**Comment by thescouser89** [comment]
LGTM!

File: guide/build-config.md (hunk 1 above)
Comment (rnc) [reply guide/build-config.md:81]: I think it should mention what the default is and what the command line flag is.

File: guide/build-config.md [hunk 2]
```diff
@@ -69,13 +70,16 @@ milestone: DR1
...
 
 The `releaseStorageUrl` is location of the release storage, typically on rcm-guest staging. This is the url to be used in the 'upload to candidates script' when running the `pig release` command. Using `*` in the milestone means that this property may also be used for automatic increment of milestone value. For example `ER*` as milestone, during `pig run`, the command will check the configured release storage and increment the version to next available milestone version. It is possible to configure it either within the `build-config.yaml` or via the CLI using the flag `--releaseStorageUrl=<...>` with the former taking precedence.
 
 The `brewTagPrefix` is an **optional** key that you can use to override the default generated brew tag prefix associated with a product version. The key is used to specify to which Brew tag all the builds should go when the milestone is closed.
 
+The `temporaryBuildAlignmentPreference` is an option to specify alignment preference for temporary build to be run with: PERSISTENT or TEMPORARY(default). This property will override --tempBuildAlignment=PERSISTENT/TEMPORARY flag passed in as command line option.
```
Comment (rnc) [inline guide/build-config.md:81]:
```suggestion
The `temporaryBuildAlignmentPreference` is an option to specify alignment preference for temporary build to be run with: PERSISTENT or TEMPORARY(default). This property will override `--tempBuildAlignment=PERSISTENT|TEMPORARY` flag passed in as command line option.
```

**Review by rnc — APPROVED** [review]

**Pull request merged**